src/event/mod.rs

```diff
@@ -32,7 +32,7 @@ impl EventSemantics for Event {
         match self.event_data {
             EventData::Icp(_) => {
                 // ICP events require the state to be uninitialized
-                if state != IdentifierState::default() || self.sn != 0 {
+                if state.prefix != IdentifierPrefix::default() || self.sn != 0 {
```

Comment:
`IdentifierState.last` was changed earlier in `SignedEventMessage::apply_to`, so it's no longer default. I'm not sure if new condition is sufficient.

Reply:
this condition should still be sufficient, however `SignedEventMessage::apply_to` could be written with `IdentifierState { last: self.serialize()?, ..self.apply_to(state)? }` which would allow the same condition to be used here but may be less readable